os/fs/driver/mtd/smart.c

```diff
@@ -3614,7 +3614,6 @@ static uint16_t smart_findfreephyssector(FAR struct smart_struct_s *dev, uint8_t
 	if (physicalsector == 0xFFFF || physicalsector >= dev->totalsectors) {
 		if (bitflipped) {
 			fdbg("retry allocation \n");
-			canrelocate = FALSE;
```

Comment:
Currently canrelocate is always false because when it called, everywhere calling this function with false value.
But considering expandability, we'd better keep this line because it checking bit flip and do not want relocate.

Reply:
ok